frontend/packages/console-shared/src/components/utils/with-resources.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { connect } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+
+import { Firehose, inject } from '@console/internal/components/utils';
+import { K8sKind, K8sResourceKindReference } from '@console/internal/module/k8s';
+
+type ResourceToPropsResult = {
+  value: {
+    [key: string]: any;
+  };
+};
+
+type ResourcesProps = {
+  loaderComponent?: React.ComponentType;
+  onError?(): void;
+  resourceMap: any;
+  resources?: any;
+  resourceToProps?(): ResourceToPropsResult;
+  children: React.ReactNode;
+};
+
+type ResourcesState = {
+  childrenProps: any;
+  errors: any[];
+  loaded: boolean;
+};
+
+function checkErrors(resources, onError): void {
+  if (resources.length > 0) {
+    if (onError) {
+      onError(resources);
+    }
+    resources.forEach(resource => {
+      const errorMessage = _.get(
+        resource.error,
+        'json.message',
+        `Error occured while loading ${resource.resourceConfig.resource.kind}`
+      );
+      const errorCode = _.get(resource.error, 'json.code', '');
+      const error = errorCode ? `${errorCode}: ${errorMessage}` : errorMessage;
+      console.warn(error); // eslint-disable-line
+    });
+  }
+}
+
+function getResourcesState({ resourceMap, resources = {}, resourceToProps = null }): ResourcesState {
+  const childrenProps = {};
+  const errors = [];
+  let loaded = true;
+
+  Object.keys(resourceMap).forEach(resourceKey => {
+    const resourceConfig = resourceMap[resourceKey];
+    const configResource = resourceConfig.resource;
+    const resource = _.get(resources, resourceKey);
+
+    if (resource) {
+      if (resource.loaded) {
+        childrenProps[resourceKey] = resource.data;
+      } else if (resourceConfig.required) {
```

Comment:
Would it make sense to do small refactoring when already moving the code?
I like `isRequired` more, as it is boolean.